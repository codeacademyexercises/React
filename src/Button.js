import React, { Component } from 'react';
// import logo from './logo.svg';
// import './App.css';

class Button extends Component {
  render() {
    return (
      <div>
          <button type="button" onClick={this.props.onClick}>{this.props.name}</button>
      </div>
    );
  }
}

export default Button;
