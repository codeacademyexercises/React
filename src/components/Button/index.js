import React, { Component } from 'react';
// import logo from './logo.svg';
// import './App.css';

const Button = (props)=> {
  return (
      <div>
          <button type="button" onClick={props.onClick}>{props.name}</button>
      </div>
    );
}

export default Button;
