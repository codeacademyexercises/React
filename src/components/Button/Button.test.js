import React from 'react';
// import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import {shallow} from 'enzyme';
import Button from '.';
import {shallow} from 'enzyme';

describe('Button ',()=>{
    it('renderer should display button',()=>{
        const tree = renderer.create(<Button/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should trigger onClick on clicking the button',()=>{
        const wrapper = shallow(<Button name="testing" onClick={jest.fn()}/>);
        wrapper.find('button').simulate('click');
        expect(wrapper.props().onClick).toHaveBeenCalled();
    });
});