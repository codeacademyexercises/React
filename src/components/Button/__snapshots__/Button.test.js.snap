// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button  renderer should display button 1`] = `
<div>
  <button
    type="button"
  />
</div>
`;
